src/components/tl/ucp/allreduce/allreduce.h

```diff
@@ -43,7 +33,6 @@ ucc_status_t ucc_tl_ucp_allreduce_init(ucc_tl_ucp_task_t *task);
     } while (0)
 
 #define ALLREDUCE_TASK_CHECK(_args, _team)                                     \
-    CHECK_USERDEFINED_OP((_args), (_team));                                    \
```

Comment:
where is the check if user-defined "op" is provided for user-defined data types?

Reply:
we don't need the check for user-defined op because now user-defined op is defined as part of user-defined type (flag on generic dt). So it is done during dt_reduce: https://github.com/openucx/ucc/pull/348/files#diff-1c664cedc0cc7a5a15ac577b975d287012b301de57fb55b92015a6842629d4d2R129

I check for generic dt and this always implies generic ops. This way internally i have only 1 runtime check in algorithms

Reply:
just checking if there is param validation check if reduce.op and reduce.context is there for  UCC_GENERIC_DT_OPS_FLAG_REDUCE

Reply:
Oh i see: you mean when running ucc_dt_reduce ? I can add the check but it would be only to detect user bugs. The reason is: if user launches collective with reduction (allreduce, reduce, reduce_scatter(v)) and passes generic dtype into it then according to UCC spec user MUST provided custom op as well attached to the custom dtype. Otherwise, we can't get to ucc_dt_reduce with generic dt.

Reply:
@bureddy is that what you meant?

Reply:
yes. I meant to check the param valiation as per spec i.e is user provided custom op with custom dtype or not

Reply:
after discussion: we guard user for all the cases, e.g. if user sets REDUCE flag on generic dt but forgets to set reduce.cb and leaves garbage there. so, leaving as is.